lib/influxer/metrics/metrics.rb

```diff
@@ -162,10 +170,50 @@ def tags
       @attributes.select { |k, _| tag_names.include?(k.to_s) }
     end
 
-    attributes :timestamp
+    def time(parse = false)
+      return unless @time
+
+      return parse_timestamp(@time) if parse
+      @time
+    end
+
+    def time=(val)
+      if val.is_a?(Date) || val.is_a?(Time)
+        @time = parse_time(val)
+      else
+        @time
+      end
+    end
 
     private
 
+    def parse_timestamp(val)
+      case client.time_precision
+      when 'ms'
```

Comment:
I know, but I believe in most cases should use seconds or milliseconds. Please read comment above.

I am making changes in the query to support epoch. Currently only supported will millisecond.